cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -138,4 +316,470 @@ private String getProgramScheduleId(Id.Program program, SchedulableProgramType p
+              lastNotification = new StreamSizeNotification(streamSize.getTimestamp(), streamSize.getSize());
+            }
+          }
+        } catch (IOException e) {
+          // In case polling the stream events size failed, we can initialize the schedule task to the last notification
+          // info if it exists, or to 0. This won't be very accurate, but this is the best info we have at that time
+          synchronized (lastNotificationLock) {
+            if (lastNotification != null) {
+              newTask.startSchedule(lastNotification.getSize(), lastNotification.getTimestamp(), active, persist);
+            } else {
+              newTask.startSchedule(0L, System.currentTimeMillis(), active, persist);
+            }
+          }
+        }
+      } else {
+        newTask.startSchedule(baseRunSize, baseRunTs, active, persist);
+      }
+
+      if (lastNotification != null) {
+        // In all cases, when creating a schedule - either if it comes from the store or not,
+        // we want to pass it the last seen notification if it exists.
+        // This call will send the notification to all the active tasks. The ones which are active before
+        // that method is called will therefore see the notification twice. It is fine though.
+        received(lastNotification, null);
+      }
+      return true;
+    }
+
+    /**
+     * Suspend a scheduling task that is based on the data received by the stream referenced by {@code this} object.
+     */
+    public synchronized void suspendScheduleTask(Id.Program programId, SchedulableProgramType programType,
+                                                 String scheduleName) throws ScheduleNotFoundException {
+      String scheduleId = getScheduleId(programId, programType, scheduleName);
+      StreamSizeScheduleTask task = scheduleTasks.get(scheduleId);
+      if (task == null) {
+        throw new ScheduleNotFoundException(scheduleName);
+      }
+      if (task.suspend()) {
+        activeTasks.decrementAndGet();
+      }
+    }
+
+    /**
+     * Resume a scheduling task that is based on the data received by the stream referenced by {@code this} object.
+     */
+    public void resumeScheduleTask(Id.Program programId, SchedulableProgramType programType, String scheduleName)
+      throws ScheduleNotFoundException {
+      int activeTasksNow;
+      StreamSizeScheduleTask task;
+      synchronized (this) {
+        String scheduleId = getScheduleId(programId, programType, scheduleName);
+        task = scheduleTasks.get(scheduleId);
+        if (task == null) {
+          throw new ScheduleNotFoundException(scheduleName);
+        }
+        if (!task.resume()) {
+          return;
+        }
+        activeTasksNow = activeTasks.incrementAndGet();
+      }
+      if (activeTasksNow == 1) {
+        // There were no active tasks until then, that means polling the stream was disabled.
```

Comment:
There is no implicit disabling of the polling. This is what happens:

- Last task is suspended, there is 0 active tasks
- The runnable created in ``createPollingRunnable`` is called, but it does nothing since the condition ``if (activeTasks.get() > 0)`` is not met.
- The ``received`` method of this StreamSubscriber is hence not called, which means ``cancelPollingAndScheduleNext`` will not be called again, and we don't reschedule the polling

So exactly one polling should happen after suspending all tasks.